tests/E2E Tests/WebAppUiTests/TestingWebAppLocally.cs

```diff
@@ -18,8 +18,7 @@ namespace WebAppUiTests;
 #if !FROM_GITHUB_ACTION && !AZURE_DEVOPS_BUILD
 
 // since this test changes environment variables we'd prefer it not run at the same time as other tests
-[CollectionDefinition(nameof(TestingWebAppLocally), DisableParallelization = true)]
-[Collection("WebAppUiTests")]
+[CollectionDefinition(nameof(UiTestNoParallelization), DisableParallelization = true)]
```

Comment:
How does this test change env variables? I am seeing a list of variables that are injected in the processes you start, but each process will maintain that. 

if the StartUrl is different (or can be made different), can you actually run some of these tests in parallel?

Reply:
I can see how the comment lacks specificity, so I'll clean that up. Essentially, because we are launching multiple apps from the same environment that need to listen on different ports, it is the Kestrel:Endpoints:Http:Url that needs to change for each process that is started. This test only has one process so you don't see it getting set independently like in the other tests, however it does set this variable in the background which can cause a race condition when run in parallel with the other UI tests.

Reply:
You can still define a method that runs in a critical section (so only 1 thread) that starts the process to ensure there is no race condition on the env variables. 

That said, env variables seem complex as inter-process communication. Why not just start each processs with some command line arguments? 

note: i realize this may be outside the scope of this PR, so if needed pls open a new work item around test optimization.